server/src/main/java/org/elasticsearch/cluster/block/ClusterBlocks.java

```diff
@@ -235,20 +228,27 @@ public ClusterBlockException indicesBlockedException(ClusterBlockLevel level, St
      * like the deletion of an index to free up resources on nodes.
      * @param indices the indices to check
      */
+
     public ClusterBlockException indicesAllowReleaseResources(String[] indices) {
-        final Function<String, Stream<ClusterBlock>> blocksForIndexAtLevel = index ->
-            blocksForIndex(ClusterBlockLevel.METADATA_WRITE, index).stream();
-        Stream<ClusterBlock> blocks = concat(
-            global(ClusterBlockLevel.METADATA_WRITE).stream(),
-            Stream.of(indices).flatMap(blocksForIndexAtLevel)).filter(clusterBlock -> clusterBlock.isAllowReleaseResources() == false);
-        Set<ClusterBlock> clusterBlocks = unmodifiableSet(blocks.collect(toSet()));
-        if (clusterBlocks.isEmpty()) {
+        Set<ClusterBlock> globalBlocks = global(ClusterBlockLevel.METADATA_WRITE);
+        Map<String, Set<ClusterBlock>> indexLevelBlocks = new HashMap<>();
+        for (String index : indices) {
+            Set<ClusterBlock> blocks = Sets.union(globalBlocks, blocksForIndex(ClusterBlockLevel.METADATA_WRITE, index))
+                .stream().filter(clusterBlock -> clusterBlock.isAllowReleaseResources() == false).collect(toSet());
+            if (blocks.isEmpty() == false) {
+                indexLevelBlocks.put(index, Sets.union(globalBlocks, blocks));
+            }
+        }
+        if (indexLevelBlocks.isEmpty()) {
+            if(!globalBlocks.stream()
+                .filter(clusterBlock -> clusterBlock.isAllowReleaseResources() == false).collect(toSet()).isEmpty()){
+                return new ClusterBlockException(globalBlocks);
```

Comment:
This is not correct. We should use the global blocks after filtered. Also, can we do this one at line 233?

Reply:
good catch